src/imgui/ImGuiMachine.cc

```diff
@@ -279,6 +267,99 @@ void ImGuiMachine::showMenu(MSXMotherBoard* motherBoard)
+						if (auto* info = findMachineInfo(currentDefault)) {
+							return info->displayName;
+						}
+						return currentDefault;
+					}();
+					im::Combo("##defaultMachine", currentDefaultDisplay.c_str(), [&]{
+						for (auto& info: allMachines) {
+							bool ok = getTestResult(info).empty();
+							im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+								if (ImGui::Selectable(info.displayName.c_str())) {
+									reactor.getDefaultMachineSetting().setValue(TclObject(info.configName));
+									// clear the default setup setting to avoid loading that at startup
+									reactor.getDefaultSetupSetting().setValue(TclObject(""));
+								}
+								if (info.configName == currentDefault && ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+								if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+									im::ItemTooltip([&]{
+										printConfigInfo(info);
+									});
+								}
+							});
+						}
+					});
+				});
+			});
+			ImGui::RadioButton("Load a setup", &startType, StartType::SETUP);
+			HelpMarker("Select this option to load the specified setup when openMSX starts up. "
+				"Note that in case the setup cannot be loaded, the machine shown at the other "
+				"option will be loaded after all.");
+			im::Indent([&] {
+				im::Disabled(startType != StartType::SETUP, [&] {
+					if (!setupSettingsOpen) {
+						setups = Reactor::getSetups();
+					}
+					auto currentDefault = std::string(reactor.getDefaultSetupSetting().getString());
+					/* TODO: debug and improve this:
+					auto currentSaveAtExitName = manager.getReactor().getSaveSetupAtExitNameSetting().getString();
+					if (!contains(setups, currentSaveAtExitName)) {
+						setups.emplace_back(currentSaveAtExitName);
+					}
+					*/
+					im::Combo("##defaultSetup", currentDefault.c_str(), [&]{
+						for (auto& setup: setups) {
+							if (ImGui::Selectable(setup.c_str())) {
+								reactor.getDefaultSetupSetting().setValue(TclObject(setup));
+							}
+							if (setup == currentDefault && ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+							if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+								if (previewSetup.name != setup) {
+									previewSetup.fullName = userDataFileContext(Reactor::SETUP_DIR).resolve(tmpStrCat(setup, Reactor::SETUP_EXTENSION));
+									loadPreviewSetup();
+								}
+								im::ItemTooltip([&]{
+									if (previewSetup.motherBoard) {
+										// TODO: add parameter to hide treenodes? Make no sense in a tooltip...
+										showSetupOverview(*previewSetup.motherBoard);
+									} else {
+										showNonExistingPreview();
+									}
+								});
+							}
+						}
+					});
+				});
+			});
+			ImGui::Separator();
+			ImGui::TextUnformatted("When openMSX exits:");
+			im::Indent([&] {
+				ImGui::TextUnformatted("Save setup level");
```

Comment:
This prints `Save setup level` and `as` *above* the selector widget. For all other widgets we put the label *after* the widget.

Maybe rename `as` to `Save setup name`.

Reply:
I think it looks clearer like this, as it tells you exactly what it will do:

![image](https://github.com/user-attachments/assets/b7f4a7f1-9457-4fbd-bf80-0fa7e3192d8d)

Reply:
Currently in the "Save setup" submenu we (almost) have this arrangement (difference: replace "save" button with "Save setup name" label). And IMHO it's equally clear.

But more important:
* Then it's consistent with how we do it everywhere else in the GUI.
* It's the default/recommended Dear ImGui way of doing this. Most widgets have builtin support for displaying a label. So the code becomes simpler.
* If you look at other applications using Dear ImGui (once you know what to look for, you notice them more and more), they all also use this pattern.

Reply:
In this particular case I found it clearer to show it as a behavior: "When openMSX exists: Save setup level Machine as last_used". That makes it very clear what the whole thing is for, by reading it as a sentence like that.
It's quite different than just an action with a button as in the Save Setup menu.